demos/imagenet_classification_demo/include/my_functor.hpp

```diff
@@ -0,0 +1,79 @@
+#include <cstdio>
+#include <string>
+#include <functional>
+#include <atomic>
+
+#include <inference_engine.hpp>
+
+#include <ie_iextension.h>
+#include <ext_list.hpp>
+
+class  InferRequestCallback
+{
+    public:
+    InferenceEngine::InferRequest& ir;
+    mutable int firstIndex;
+
+    std::mutex&  mutex;
+    std::condition_variable& condVar;
+    std::vector<cv::Mat>& inputImgs;
+    std::string inputBlobName;
+    unsigned batchSize;
+    std::queue<cv::Mat>& showMats;
+    unsigned& curPos;
+    unsigned& framesNum;
+    std::atomic<bool>& quitFlag;
+
+    InferRequestCallback(InferenceEngine::InferRequest& ir,
+                int index, 
+                std::mutex& mutex, std::condition_variable& condVar,
+                std::vector<cv::Mat>& inputImgs,
+                const std::string& inputBlobName,
+                unsigned batchSize,
+                std::queue<cv::Mat>& showMats,
+                unsigned& curPos,
+                unsigned& framesNum,
+                std::atomic<bool>& quitFlag
+                ):
+                ir(ir), firstIndex(index),
+                mutex(mutex), condVar(condVar),
+                inputImgs(inputImgs),
+                inputBlobName(inputBlobName),
+                batchSize(batchSize),
+                showMats(showMats),
+                curPos(curPos),
+                framesNum(framesNum),
+                quitFlag(quitFlag) {}
+
+
+    //curPos and framesNum is stored in ieWrapper
+    void operator()() const{
+        if(!quitFlag) {
+            int inputDataSize = inputImgs.size();
+            {
```

Comment:
This line should be merged with previous one.

Reply:
Done